### PR TITLE
Use reference quantization on CPU if compiled extensions are unavailable

### PR DESCRIPTION
**Pull request by vshampor**
### Changes
As stated in the title. Also changed the way of loading extensions to the lazy approach, so that the extensions are only loaded if associated quantization or binarization functionality is invoked, meaning that non-INT8 workflows will have less points of failure for the user.

### Reason for changes
UX improvement due to not failing where we may keep working functionally.

### Related tickets
94348

### Tests
test_extension_unavailable

### CODE REVIEW

File: tests/shared/helpers.py [hunk 1]
```diff
@@ -28,6 +28,10 @@
 from tests.shared.paths import PROJECT_ROOT
 
 TensorType = TypeVar('TensorType')
+TEST_ROOT = Path(__file__).absolute().parents[1]  # https://github.com/PyCQA/pylint/issues/5783
+PROJECT_ROOT = TEST_ROOT.parent.absolute()  # pylint:disable=no-member
+EXAMPLES_DIR = PROJECT_ROOT / 'examples'
+GITHUB_REPO_URL = 'https://github.com/openvinotoolkit/nncf/'
```
Comment (ljaljushkin) [inline tests/shared/helpers.py:34]: Is it used somewhere?

File: tests/torch/quantization/test_functions.py [hunk 2]
```diff
@@ -96,39 +47,51 @@ def _seed():
...
+                               bits: int,
+                               get_deviation: Callable[[], int],
+                               min_deviation: float):
+    if np.prod(input_low.shape) == 1:
+        input_low_ch, input_range_ch = \
+            map(lambda x: x.squeeze(), [input_low, input_range])
     else:
-        min_adj_ch, input_low_ch, input_range_ch, quant_len_ch =\
-            map(lambda x: x[tuple(ch_idx)].squeeze(), [min_adj, input_low, input_range, quant_len])
+        input_low_ch, input_range_ch = \
```
Comment (ljaljushkin) [inline tests/torch/quantization/test_functions.py:64]: Is it equivalent without `min_adj_ch` and `quant_len`? Why was it needed before?

File: tests/torch/quantization/test_functions.py [hunk 3]
```diff
@@ -242,7 +207,7 @@ def to_tensor(a):
 
 
 def check_outputs_for_quantization_functions(test_val: torch.Tensor, ref_val: np.ndarray,
-                                             is_fp16: bool, rtol=1e-4, atol=1e-10):
```
Comment (ljaljushkin) [inline tests/torch/quantization/test_functions.py:245]: Was `is_fp16` unused before?

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: nncf/torch/binarization/binarize_functions.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 
 from nncf.common.logging import nncf_logger
 from nncf.torch.utils import add_domain
+from nncf.torch.binarization.extensions import BinarizedFunctionsCUDA
 
 from .extensions import BinarizedFunctionsCUDA
```
[the remark this answers is not in the document]
Comment (vshampor) [reply nncf/torch/binarization/binarize_functions.py:22]: Thanks, will remove

File: tests/shared/helpers.py (hunk 1 above)
Comment (vshampor) [reply tests/shared/helpers.py:34]: Indeed, in `tests.shared.helpers.create_venv_with_nncf` when we try to install from a commit on github.

File: tests/torch/quantization/test_functions.py (hunk 2 above)
Comment (vshampor) [reply tests/torch/quantization/test_functions.py:64]: No idea, the original code was by @daniil-lyakhov. I reviewed the usage of these variables carefully and saw no reason to keep them, they are equivalent to some arguments already in the scope. Maybe @daniil-lyakhov just forgot to remove them.

File: tests/torch/quantization/test_functions.py (hunk 3 above)
Comment (vshampor) [reply tests/torch/quantization/test_functions.py:245]: No, even my IDE marked this as an unused parameter.

**Review by alexsu52 — APPROVED** [review]
LGTM

File: tests/shared/helpers.py (hunk 1 above)
Comment (ljaljushkin) [reply tests/shared/helpers.py:34]: I can't see the usage of `GITHUB_REPO_URL` in your PR. 
The same 4 lines are defined in the tests/shared/paths.py
Why not reuse them from this file?

File: nncf/torch/binarization/binarize_functions.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (vshampor) [reply nncf/torch/binarization/binarize_functions.py]: Done

File: tests/shared/helpers.py (hunk 1 above)
Comment (vshampor) [reply tests/shared/helpers.py:34]: I get it now, that was a rebase artifact. Removed.

**Review by ljaljushkin — APPROVED** [review]

File: tests/torch/quantization/test_functions.py (hunk 2 above)
Comment (daniil-lyakhov) [reply tests/torch/quantization/test_functions.py:64]: Yeah, for previous version of the quantize function this `min_adj_ch` was not equal to `-inp_low`. So, this changes are totally valid

**Review by daniil-lyakhov — APPROVED** [review]

**Pull request merged**